Test failure - test_segment_from_mask_non_square

There is a failing test: test_segment_from_mask_non_square

It fails when you run the tests from either the master or dev branches.
* [Here are the results from the master branch](https://github.com/computational-cell-analytics/micro-sam/actions/runs/7722147756)
* [Same results from the dev branch](https://github.com/computational-cell-analytics/micro-sam/actions/runs/7722146152)

Here are the relevant sections of test code:
* [the test that fails](https://github.com/computational-cell-analytics/micro-sam/blob/8151561ea005331ed34b5d691c7440e79db4db7c/test/test_prompt_based_segmentation.py#L106-L107)
* [the actual part that runs](https://github.com/computational-cell-analytics/micro-sam/blob/8151561ea005331ed34b5d691c7440e79db4db7c/test/test_prompt_based_segmentation.py#L52-L101)
* [the line reported as failing](https://github.com/computational-cell-analytics/micro-sam/blame/8151561ea005331ed34b5d691c7440e79db4db7c/test/test_prompt_based_segmentation.py#L77)

This is the error message:

 FAILED test/test_prompt_based_segmentation.py::TestPromptBasedSegmentation::test_segment_from_mask_non_square - AssertionError: 0.012674722835867252 not greater than 0.8


```python-traceback
=================================== FAILURES ===================================
________ TestPromptBasedSegmentation.test_segment_from_mask_non_square _________

self = <test.test_prompt_based_segmentation.TestPromptBasedSegmentation testMethod=test_segment_from_mask_non_square>

    def test_segment_from_mask_non_square(self):
>       self._test_segment_from_mask((256, 384))

test/test_prompt_based_segmentation.py:107: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
test/test_prompt_based_segmentation.py:77: in _test_segment_from_mask
    self.assertGreater(util.compute_iou(mask, predicted), expected_iou_mask)
E   AssertionError: 0.012674722835867252 not greater than 0.8
```

--- a/micro_sam/prompt_based_segmentation.py
+++ b/micro_sam/prompt_based_segmentation.py
@@ -6,6 +6,7 @@ import warnings
 from typing import Optional, Tuple
 
 import numpy as np
+import torch
 from nifty.tools import blocking
 from skimage.feature import peak_local_max
 from skimage.filters import gaussian
@@ -87,12 +88,14 @@ def _compute_logits_from_mask(mask, eps=1e-3):
 
     elif logits.shape[0] == logits.shape[1]:  # shape is square
         trafo = ResizeLongestSide(expected_shape[0])
-        logits = trafo.apply_image(logits[..., None])
+        logits = trafo.apply_image_torch(torch.from_numpy(logits[None, None]))
+        logits = logits.numpy().squeeze()
 
     else:  # shape is not square
         # resize the longest side to expected shape
         trafo = ResizeLongestSide(expected_shape[0])
-        logits = trafo.apply_image(logits[..., None])
+        logits = trafo.apply_image_torch(torch.from_numpy(logits[None, None]))
+        logits = logits.numpy().squeeze()
 
         # pad the other side
         h, w = logits.shape

